### PR TITLE
72: Adds TypeORM

### LINKED ISSUE #72: integrate DB client
we can either use a full ORM, or a lighter-weight driver/driver + query builder plus some migration tool

Driver:
- https://node-postgres.com/

Query builder:
- http://knexjs.org/

ORMs:
- https://github.com/typeorm/typeorm,  https://github.com/typeorm/typescript-express-example
- https://sequelize.org/master/manual/typescript.html

migration tools:
- https://www.npmjs.com/package/db-migrate

### PR DESCRIPTION
**Pull request by jlhogan**
Closes #72

With `migrationsRun`: true, app creates these tables on startup:
```
postgres-# \dt
                List of relations
 Schema |         Name         | Type  |  Owner
--------+----------------------+-------+----------
 public | child                | table | postgres
 public | enrollment           | table | postgres
 public | family               | table | postgres
 public | funding              | table | postgres
 public | funding_space        | table | postgres
 public | funding_time_split   | table | postgres
 public | income_determination | table | postgres
 public | migrations           | table | postgres
 public | organization         | table | postgres
 public | permission           | table | postgres
 public | report               | table | postgres
 public | reporting_period     | table | postgres
 public | site                 | table | postgres
 public | user                 | table | postgres
```

### CODE REVIEW

File: src/index.ts.gen [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import "reflect-metadata";
```
Comment (jlhogan) [inline src/index.ts.gen:1]: delete

File: tsconfig.json.gen [hunk 2]
```diff
@@ -0,0 +1,15 @@
+{
```
Comment (jlhogan) [inline tsconfig.json.gen:1]: delete

File: src/entity/Child.ts [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+  hispanicOrLatinxEthnicity?: boolean;
+
+  @Column({ nullable: true })
+  gender?: Gender;
+
+  @Column({ nullable: true })
+  foster?: boolean;
+
+  @Column({ default: false })
+  recievesC4K: boolean = false;
```
Comment (jlhogan) [inline src/entity/Child.ts:71]: matches new data collection requirements; c4k boolean over familyID + cert dates

File: src/entity/Family.ts [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+  state?: string;
+
+  @Column({ nullable: true })
+  zip?: string;
+
+  @Column({ nullable: true })
+  homelessness?: boolean;
+
+  @OneToMany((type) => IncomeDetermination, (det) => det.family)
+  incomeDeterminations: Array<IncomeDetermination>;
```
Comment (jlhogan) [inline src/entity/Family.ts:38]: changed the name of this relation (determinations -> incomeDeterminations), and the related model (FamilyDetermination -> IncomeDetermination) b/c it had been annoying me :)

File: src/entity/Report.ts [hunk 5]
```diff
@@ -0,0 +1,57 @@
+import {
```
Comment (jlhogan) [inline src/entity/Report.ts:1]: do we even need a report model at this stage?

**Comment by jlhogan** [comment]
overall this ORM is p great, but im just wondering if we need this whole complex data model. Makes validations easier, but do we really need all these discrete entity types?

File: README.md [hunk 6]
```diff
@@ -30,8 +30,8 @@ To generate the client code, run the included Bash file: `./generate-client-code
 1. Install (if you haven't already) [Docker](https://hub.docker.com/search?q=&type=edition&offering=community). Make sure you have [Docker Compose](https://docs.docker.com/compose/install/), which is included in some OS distributions but must be installed separately in others.
 2. Install all corresponding yarn dependencies, based on the static versions specified in `yarn.lock`, via the containers:
     ```sh
-    docker-compose --entrypoint "yarn install --frozen-lockfile" client
-    docker-compose --entrypoint "yarn install --frozen-lockfile" server
+    docker-compose exec client yarn install --frozen-lockfile
```
Comment (JordanGuinn) [inline README.md:33]: 👍🏽👍🏽👍🏽👍🏽👍🏽👍🏽👍🏽👍🏽

File: package.json [hunk 7]
```diff
@@ -39,7 +42,9 @@
     "watch-debug": "concurrently -k -p \"[{name}]\" -n \"TypeScript,Node\" -c \"cyan.bold,green.bold\" \"npm run watch-ts\" \"npm run serve-debug\"",
     "watch-node": "nodemon dist/index.js",
     "watch-test": "npm run test -- --watchAll",
-    "watch-ts": "tsc -w"
+    "watch-ts": "tsc -w",
+    "prettier": "prettier --single-quote --write **/*.{js,jsx,ts,tsx,json,css,scss}",
```
Comment (JordanGuinn) [inline package.json:46]: 🔥🔥 Does this mean we can update our pre-commit git hook to something like `docker-compose exec -T server "yarn run prettier"`?!

File: src/entity/Child.ts [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+
+import { Enrollment } from './Enrollment';
+import { Family } from './Family';
+import { Gender } from './enums/Gender';
+import { Organization } from './Organization';
+
+@Entity()
+export class Child {
+  @PrimaryGeneratedColumn()
+  id: string;
```
Comment (JordanGuinn) [inline src/entity/Child.ts:17]: Does `@PrimaryGeneratedColumn()` actually generate a string?  I don't know much about Postgres, but my suspicion is that it'd gen an auto-incrementing integer

File: src/entity/Permission.ts [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+import { Organization } from './Organization';
+import { Site } from './Site';
+
+enum PermissionType {
+  Organization = 1,
+  Site = 2,
+}
+
+@Entity()
+@TableInheritance({ column: { type: 'enum', enum: PermissionType } })
```
Comment (JordanGuinn) [inline src/entity/Permission.ts:18]: 🔥

File: src/entity/Child.ts (hunk 8 above)
Comment (jlhogan) [reply src/entity/Child.ts:17]: ah yes! this was still a "todo" -- there is a UUID generator option; i forgot to follow up on this. todaY :tada:

**Pull request merged**